components/range_cache_memory_engine/src/range_manager.rs

```diff
@@ -479,6 +482,19 @@ impl RangeManager {
     pub fn get_and_reset_range_evictions(&self) -> u64 {
         self.range_evictions.swap(0, Ordering::Relaxed)
     }
+
+    // Only ranges that have not been scheduled will be retained in `ranges`
+    pub fn schedule_ranges_to_delete(&mut self, ranges: &mut Vec<CacheRange>) {
```

Comment:
```suggestion
    pub fn mark_delete_ranges_scheduled(&mut self, ranges: &mut Vec<CacheRange>) {
```

It does not schedule ranges but rather marks them as scheduled.